main.go

```diff
@@ -30,42 +31,69 @@ func init() {
 	httpcaddyfile.RegisterHandlerDirective("dynamic_proxy", parseCaddyfile)
 }
 
+type AuthSites map[string]AuthSite
+
+func (a *AuthSites) Decode(input string) error {
+	*a = make(AuthSites)
+	sites := strings.Split(input, ";")
+	for _, site := range sites {
+		if site == "" {
+			break
+		}
+
+		level, site, found := strings.Cut(site, ":")
+		if !found {
+			return fmt.Errorf("unable to decode env variable: %v", level)
+		}
+
+		var as AuthSite
+		if err := as.Decode(site); err != nil {
+			return err
+		}
+
+		(*a)[level] = as
+	}
+
+	return nil
+}
+
 type AuthSite struct {
 	To   string
 	Path string
 }
 
-func NewAuthSite(site string) (AuthSite, error) {
-	// Add protocol if it is missing (needed for url parse)
-	if m, _ := regexp.MatchString("^\\w+://", site); !m {
-		site = "http://" + site
+func (a *AuthSite) Decode(input string) error {
+	if input == "" {
+		return fmt.Errorf("cannot decode empty string")
 	}
 
-	u, err := url.Parse(site)
+	re, err := regexp.Compile(`^\w+://`)
 	if err != nil {
-		return AuthSite{}, err
+		return err
 	}
 
-	result := AuthSite{
-		To:   u.Host,
-		Path: u.Path,
+	// Add protocol if it is missing (needed for url parse)
+	if !re.MatchString(input) {
+		input = "http://" + input
```

Comment:
Is this safe enough or would it be better just to return an error and say that the protocol is required?
I'm just wondering if in situations where it should really be `https` that the resulting problem/failure will be hidden from us.

Reply:
The protocol is currently not used, it is just there so url.Parse can parse it. I just get the host and path.